hist/hist/inc/TH1.h

```diff
@@ -137,6 +212,158 @@ class TH1 : public TNamed, public TAttLine, public TAttFill, public TAttMarker {
+
+      auto configureBins = [&](auto &hist) {
+         if constexpr (ndim == 1) {
+             hist.SetBins(nBins[0], binLowEdge[0], binUpEdge[0]);
+         } else if constexpr (ndim == 2) {
+             hist.SetBins(nBins[0], binLowEdge[0], binUpEdge[0],
+                          nBins[1], binLowEdge[1], binUpEdge[1]);
+         } else if constexpr (ndim == 3) {
+             hist.SetBins(nBins[0], binLowEdge[0], binUpEdge[0],
+                          nBins[1], binLowEdge[1], binUpEdge[1],
+                          nBins[2], binLowEdge[2], binUpEdge[2]);
+         }
+      };
+      configureBins(*hSlice);
+
+      std::vector<Double_t> sliceContents;
+      std::unordered_set<Int_t> processedBins;
+      Double_t underflowContent = 0.0, overflowContent = 0.0;
+
+      auto processBinContent = [&](Int_t x, Int_t y, Int_t z) {
+         const Int_t globalBin = GetBin(x, y, z);
+         if (!processedBins.insert(globalBin).second) return;
+ 
+         const auto content = RetrieveBinContent(globalBin);
+         const bool isUnderflow = (x < binsArray[0] || (ndim > 1 && y < binsArray[2]) || (ndim > 2 && z < binsArray[4]));
+         const bool isOverflow = (x >= binsArray[1] || (ndim > 1 && y >= binsArray[3]) || (ndim > 2 && z >= binsArray[5]));
+ 
+         if (isUnderflow) {
+             underflowContent += content;
+         } else if (isOverflow) {
+             overflowContent += content;
+         } else {
+             sliceContents.push_back(content);
+         }
+      };
+
+      for (Int_t x = 0; x <= fXaxis.GetNbins() + 1; ++x) {
+         for (Int_t y = 0; y <= (ndim > 1 ? fYaxis.GetNbins() + 1 : 1); ++y) {
+             for (Int_t z = 0; z <= (ndim > 2 ? fZaxis.GetNbins() + 1 : 1); ++z) {
+                 processBinContent(x, y, z);
+             }
+         }
+     }
+
+      unpack_caller<ndim*2> caller;
+      auto setSliceContents = [&](auto... binEdges) {
+         ROOT::Internal::SetSliceContent(*hSlice, sliceContents, binEdges...);
+      };
+      caller(setSliceContents, shiftedBins);
+         
+      hSlice->SetBinContent(0, underflowContent);
+      auto overflowBin = (ndim == 1) ? hSlice->GetBin(fXaxis.GetNbins() + 1)
+                     : (ndim == 2) ? hSlice->GetBin(fXaxis.GetNbins() + 1, fYaxis.GetNbins() + 1)
+                     : hSlice->GetBin(fXaxis.GetNbins() + 1, fYaxis.GetNbins() + 1, fZaxis.GetNbins() + 1);
+      hSlice->SetBinContent(overflowBin, overflowContent);
+
+      return hSlice.release();
+   }
+
+   template <typename T, typename... Args>
+   friend auto ROOT::Internal::Slice(const T &histo, Args... args);
+
+   template <typename... Args>
+   void SetSliceContent(const std::variant<std::vector<Double_t>, Double_t> &input, Args... args) {
+      constexpr auto ndim = sizeof...(args) / 2;
+      if (ndim != fDimension) {
+         Error("TH1::SetSliceContent", "Number of edges in the specified slice does not match histogram dimension.");
```

Comment:
Same comment as above: `Error()` + continue is a code smell: errors should interrupt the execution and return an error value (or throw an exception)